### PR TITLE
Package julia-ctags

### PR DESCRIPTION
**Pull request by m-wells**
Turn this into a proper package.  Further features if desired can be implemented but this provides the minimal functionality.  Having this in a package also means that we can ship a ctags install via [Conda.jl](https://github.com/JuliaPy/Conda.jl) or [BinaryProvider.jl](https://github.com/JuliaPackaging/BinaryProvider.jl).

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+name = "JuliaCtags"
+uuid = "1c25b4e4-f198-11e9-1949-65e68c540e8d"
+authors = []
```
Comment (oxinabox) [inline Project.toml]:
```suggestion
```

File: Project.toml [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+version = "0.1.0"
```
Comment (oxinabox) [inline Project.toml]:
```suggestion
version = "0.3.0"
```

I know there have been at least 3 major changes to the tag so might as well.

Could also just do:
```suggestion
version = "1.0.0"
```

File: README.md [hunk 3]
```diff
@@ -2,6 +2,17 @@
...
+To install run
+```julialang
+using Pkg
+Pkg.add("Interpolations")
+```
+or
+```
+] add JuliaCtags
+```
+To obtain the ctags configuration file use `get_config()` which will create a file called `julia_ctags` in the current directory.
```
Comment (oxinabox) [inline README.md]: How about we make:
```suggestion
To obtain the ctags configuration file use `write_config(dir=pwd())` which will create a file called `julia_ctags` in the targetted directory.
```

File: src/JuliaCtags.jl [hunk 4]
```diff
@@ -0,0 +1,10 @@
+module JuliaCtags
+
+export get_config
+
+function get_config(destfile = "julia_ctags")
+    sourcefile = (@__DIR__)*"/ctags"
```
Comment (oxinabox) [inline src/JuliaCtags.jl]:
```suggestion
    sourcefile = joinpath(@__DIR__, "ctags")
```

File: README.md [hunk 5]
```diff
@@ -2,6 +2,17 @@
...
+
```
Comment (oxinabox) [inline README.md:16]: Need to split this section from the below,
and maybe briefly say how the created file is used in the below process.

**Comment by oxinabox** [comment]
Ok, I am sold on this idea.
but lets not do it til we actually ship the exectuable for testing purposes at least?

**Comment by m-wells** [comment]
I think packaging it will actually make shipping an executable much easier.  Create a ctags binary using [BinaryBuilder](https://github.com/JuliaPackaging/BinaryBuilder.jl) and install the tarballs with [BinaryProvider](https://github.com/JuliaPackaging/BinaryProvider.jl).  Then when a user adds the package, the ctags executable is installed as a dependency.

File: README.md (hunk 3 above)
Comment (m-wells) [reply README.md]: How about we split the difference? `write_config(filename="./julia_ctags")`

Comment (oxinabox) [reply README.md]: sounds great to me

**Comment by ViralBShah** [comment]
I wonder if this should be merged and released as a package? Alternatively, close?

**Comment by oxinabox** [comment]
Merged.
I updated readme with how to install it as an unregistered package.
If someone else has motivation to register it they should feel free to.
Though I kinda do still think it should be made to use the ctags executable via a JLL
and then have that working (possibly as a julia app) before it is really worth registering

**Pull request merged**